cmd/thanos/sidecar.go

```diff
@@ -45,8 +45,10 @@ import (
 func registerSidecar(app *extkingpin.App) {
 	cmd := app.Command(component.Sidecar.String(), "Sidecar for Prometheus server")
 	conf := &sidecarConfig{}
+	reqLogDecision := cmd.Flag("log.request.decision", "Deprecation Warning - This flag would be soon deprecated, and replaced with request.logging. Request Logging for logging the start and end of requests. By default this flag is disabled. LogFinishCall: Logs the finish call of the requests. LogStartAndFinishCall: Logs the start and finish call of the requests. NoLogCall: Disable request logging.").Default("").Enum("NoLogCall", "LogFinishCall", "LogStartAndFinishCall", "")
```

Comment:
Why we are adding flag to deprecate it (:

Reply:
Makes sense.